core/server/services/oembed.js

```diff
@@ -53,7 +59,26 @@ class OEmbed {
      */
     constructor({config, externalRequest}) {
         this.config = config;
-        this.externalRequest = externalRequest;
+        /** @type {IExternalRequest} */
+        this.externalRequest = async (url, requestConfig) => {
```

Comment:
This wraps the external request with checks before & after for ip/localhost - this is so custom providers don't need to know/care about that. 

I want to check this looks correct, and if it does, then we should be able to remove these checks from the methods below and just use `this.externalRequest`

Reply:
👍 LGTM and agree we can remove the other checks below